experimental/modular_oak_functions/README.md

```diff
@@ -0,0 +1,75 @@
+# Proposal for making Oak Functions Extensions more general
+
+We currently have ExtensionFactories that generate Extensions. Extensions
+provide the functions available to the per-request Wasm business logic
+instances. This document proposes a generalisation of extensions and the ideas
+mentioned in issue [#2551](https://github.com/project-oak/oak/issues/2551).
+
+## Naming
+
+We have previously discussed using a more general term, such as "Service" rather
+than "Extension". My suggestion is to use `Service` as a replacement for
+`ExtensionFactory` and `ServiceProxy` as a replacement for `Extension`.
+
+The shared state used by an extension is generally owned and managed by the
+extension factory, and each extension is specific to a single `WasmState`
+instance that manages the Wasm execution for a single request. I also considered
+`ServiceFactory` and `Service` as respective replacements, but seeing that the
+`ServiceFactory` would also manage shared state it seems somewhat inaccurate.
```

Comment:
Happy with extension as the name, as long as we can answer "What are we extending?". Is it the `WasmState`?

Reply:
I don't think so, as we won't have any gRPC in the TCB.